core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -98,9 +96,8 @@ object StaticFile {
         }
       }
 
-      val hs = `Last-Modified`(lastModified) ::
-               `Content-Length`(contentLength) ::
-               contentType.toList
+      var hs: List[Header] = `Content-Length`(contentLength) :: contentType.toList
```

Comment:
```scala
val lastModifiedHeader : List[Header] = lm match {
     case None => List.empty[Header]
     case Some(httpDate) => List(`Last-Modified(httpDate)`)
}

val hs = `Content-Length`(contentLength) :: lastModifiedHeader :::  contentType.toList
```

Use property of List of ZeroOrMoreValues, or it can be handled within a single match. However should be nicer than effectful code.